src/metabase/integrations/slack.clj

```diff
@@ -46,32 +54,34 @@
 (defn- do-slack-request [request-fn endpoint request]
   (let [token (or (get-in request [:query-params :token])
                   (get-in request [:form-params :token])
+                  (slack-app-token)
                   (slack-token))]
     (when token
       (let [url     (str slack-api-base-url "/" (name endpoint))
             _       (log/trace "Slack API request: %s %s" (pr-str url) (pr-str request))
-            request (merge-with merge
-                      {:query-params   {:token token}
-                       :as             :stream
-                       ;; use a relatively long connection timeout (10 seconds) in cases where we're fetching big
-                       ;; amounts of data -- see #11735
-                       :conn-timeout   10000
-                       :socket-timeout 10000}
-                      request)]
+            request (merge-with
+                     (fn [a b] (if (and (map? a) (map? b)) (merge a b) b))
```

Comment:
`(merge-with merge ...)` (the previous implementation) throws an exception when the values associated with the conflicting key aren't both maps. This changes it to a custom function that merges sub-maps (so you could pass in additional `:headers`, for example), but otherwise just uses the latter key.

Reply:
should we use `medley.core/deep-merge` here?

Reply:
Yes, we should :) forgot that that function exists